### PR TITLE
sysadmins: Document hard-link semantic change for in-place import

### PR DESCRIPTION
**Pull request by ghost**
For https://trello.com/c/QYIXKFHK/82-hardlink-permissions-on-centos-ubuntu

/cc @manics @joshmoore

### CODE REVIEW

**Comment by hflynn** [comment]
This is quite a lot of content for a note box, I'm wondering if it would be better to just have it as a new paragraph of text.

File: omero/sysadmins/in-place-import.txt [hunk 1]
```diff
@@ -279,6 +279,23 @@ If you are unclear about how hard-linking works, please see the
 `Hard link <http://en.wikipedia.org/wiki/Hard_link>`_ article on
 Wikipedia.
 
+.. note:: The semantics of hard-linking have changed recently on Linux
+    systems with the "protected hardlinks" feature, which is enabled
+    by default and is in use on Ubuntu 14.04, CentOS 7 and other
+    contemporary systems.  Linux now requires that when you create a
+    hard-link to a file that you are either the *owner* of the file,
```
Comment (mtbc) [inline omero/sysadmins/in-place-import.txt]: I'm doubtful about the double "that"; maybe start with the "when"?

File: omero/sysadmins/in-place-import.txt [hunk 2]
```diff
@@ -279,6 +279,23 @@ If you are unclear about how hard-linking works, please see the
...
+    or that you have *read/write access* to the file.  Other Unix
+    systems, and older Linux systems, allow a hard link to be made if
```
Comment (mtbc) [inline omero/sysadmins/in-place-import.txt]: hard-link as a noun had a hyphen just above

File: omero/sysadmins/in-place-import.txt [hunk 3]
```diff
@@ -279,6 +279,23 @@ If you are unclear about how hard-linking works, please see the
...
+    you have *search access* to the file (i.e. you have read/execute
+    permission on the directory containing the file), but do not check
+    the file permissions themselves.  See the `kernel-hardening
+    mailing list post
+    <http://www.openwall.com/lists/kernel-hardening/2012/02/21/20>`__
+    which describes the change in more detail.  The implication for
+    in-place import is that the user performing the import must own or
+    have read-write permissions on the data files being imported
```
Comment (mtbc) [inline omero/sysadmins/in-place-import.txt]: it was "read/write access" above and is "read-write permissions" here, not sure if that's a useful change

**Comment by mtbc** [comment]
I'm not sure if they could possibly be attempting in-place from a system where client-side it _looks_ (due to network mount options) that they have the required permissions but in fact on the underlying filesystem they do not; can that situation arise for in-place? It's too early in the morning for me to know. But, even so, perhaps that's an awkward detail not worth adding. Also, I don't know if it's worth mentioning that symlinks should still work.

**Comment by ghost** [comment]
The suggested corrections have been made.

@mtbc Regarding systems where the client-side looks correct, I think that's a bit out of scope--we can't do much if the system is behaving in a non-standard way.  They will need to determine for themselves if hardlinks work or not.

**Comment by mtbc** [comment]
Staged at http://www.openmicroscopy.org/site/support/omero5.2-staging/sysadmins/in-place-import.html.

**Comment by mtbc** [comment]
Thank you, good to merge.

**Pull request merged**